plugins/uniques.js

```diff
@@ -557,15 +876,30 @@ var setup = function() {
 	window.plugin.uniques.setupCSS();
 	window.plugin.uniques.setupContent();
 	window.plugin.uniques.loadLocal('uniques');
-	window.addPortalHighlighter('Uniques', window.plugin.uniques.highlighter);
+	window.plugin.uniques.loadLocal('missedLatLngs');
+	window.plugin.uniques.loadLocal('parsedMsgs');
+
+	window.plugin.uniques.removeOldParsedMsgs();
+
+	window.addPortalHighlighter('Uniques: drone visited', window.plugin.uniques.highlighterDroned);
+	window.addPortalHighlighter('Uniques: scout controlled', window.plugin.uniques.highlighterScouted);
+	window.addPortalHighlighter('Uniques: visited/captured', window.plugin.uniques.highlighterCaptured);
+
 	window.addHook('portalDetailsUpdated', window.plugin.uniques.onPortalDetailsUpdated);
 	window.addHook('publicChatDataAvailable', window.plugin.uniques.onPublicChatDataAvailable);
+	window.addHook('alertsChatDataAvailable', window.plugin.uniques.onPublicChatDataAvailable);
+	window.addHook('portalAdded', window.plugin.uniques.onPortalAdded);
 	window.plugin.uniques.registerFieldForSyncing();
-	
+
 	// to mark mission portals as visited
 	window.addHook('plugin-missions-mission-changed', window.plugin.uniques.onMissionChanged);
 	window.addHook('plugin-missions-loaded-mission', window.plugin.uniques.onMissionLoaded);
-	
+
+	// add controls to toolbox
+
+	link = $("<a onclick=\"window.plugin.uniques.options();return false\" title=\"Manage UNIQUES\">Uniques</a>");
```

Comment:
```suggestion
	var link = $('<a onclick="window.plugin.uniques.options();return false" title="Manage UNIQUES">Uniques</a>');
```

BUT!
Avoid js code in html attributes.
Replace it with jQuery listeners.
```suggestion
	var link = $('<a title="Manage UNIQUES">Uniques</a>')
	  .click(window.plugin.uniques.options);
```